src/lib/parts.ts

```diff
@@ -121,7 +122,14 @@ export class AttributeCommitter {
       templatePart?: AttributeTemplatePart,
       kind: 'property'|'attribute' = 'attribute') {
     this.element = element;
-    this.name = name;
+    const colonIndex = name.indexOf(':');
+    if (colonIndex === -1) {
+      this.name = name;
+      this.namespace = null;
+    } else {
+      this.namespace = name.substring(0, colonIndex);
```

Comment:
Oh, and this is incorrect. The NS is a URI, not the `xlink` substring.

Reply:
Can you expand on that? It's incorrect in what sense?

Reply:
https://developer.mozilla.org/en-US/docs/Web/API/Element/setAttributeNS#Example

Is there a way to get the namespace URI from a prefix, right from the DOM? Otherwise this is going to be very tricky.

Reply:
The `namespace` used in `getAttributeNS` and friends is `http://www.w3.org/1999/xlink`. But each NS can be totally unique URIs in SVG, you can just register them. See the console logs in https://output.jsbin.com/mexedos/quiet

Reply:
HTML itself doesn’t have any namespaces (that do anything). SVG has `xml` and `xlink`, and any self registered namespaces don’t do anything. There’s also MathML, but it doesn’t matter without browser support.

I think you can get away with just hardcoding: https://github.com/google/incremental-dom/blob/master/src/attributes.ts#L28